nilearn/plotting/html_surface.py

```diff
@@ -0,0 +1,467 @@
+    info = {}
+    colors, cmin, cmax, cmap, norm, abs_threshold = colorscale(
+        cmap, surf_map, threshold, symmetric_cmap=symmetric_cmap)
+    info['inflated_left'] = to_plotly(surf_mesh)
+    info['vertexcolor_left'] = _get_vertexcolor(
+        surf_map, cmap, norm, abs_threshold, bg_map)
+    info["cmin"], info["cmax"] = float(cmin), float(cmax)
+    info['black_bg'] = black_bg
+    info['full_brain_mesh'] = False
+    info['colorscale'] = colors
+    return info
+
+
+def _check_mesh(mesh):
+    if not isinstance(mesh, str):
+        assert isinstance(mesh, collections.Mapping)
+        assert {'pial_left', 'pial_right', 'sulc_left', 'sulc_right',
+                'infl_left', 'infl_right'}.issubset(mesh.keys())
+        return mesh
+    mesh = datasets.fetch_surf_fsaverage(mesh)
+    return mesh
+
+
+def full_brain_info(volume_img, mesh='fsaverage5', threshold=None,
+                    cmap=cm.cold_hot, black_bg=False, symmetric_cmap=True,
+                    vol_to_surf_kwargs={}):
+    """
+    Project 3d map on cortex; prepare info to plot both hemispheres.
+
+
+    This computes the dictionary that gets inserted in the web page,
+    which contains encoded meshes, colors, min and max values, and
+    background color.
+
+    """
+    info = {}
+    mesh = _check_mesh(mesh)
+    surface_maps = {
+        h: surface.vol_to_surf(volume_img, mesh['pial_{}'.format(h)],
+                               **vol_to_surf_kwargs)
+        for h in ['left', 'right']
+    }
+    colors, cmin, cmax, cmap, norm, abs_threshold = colorscale(
+        cmap, np.asarray(list(surface_maps.values())).ravel(), threshold,
+        symmetric_cmap=symmetric_cmap)
+
+    for hemi, surf_map in surface_maps.items():
+        bg_map = surface.load_surf_data(mesh['sulc_{}'.format(hemi)])
+        info['pial_{}'.format(hemi)] = to_plotly(mesh['pial_{}'.format(hemi)])
+        info['inflated_{}'.format(hemi)] = to_plotly(
+            mesh['infl_{}'.format(hemi)])
+
+        info['vertexcolor_{}'.format(hemi)] = _get_vertexcolor(
+            surf_map, cmap, norm, abs_threshold, bg_map)
+    info["cmin"], info["cmax"] = float(cmin), float(cmax)
+    info['black_bg'] = black_bg
+    info['full_brain_mesh'] = True
+    info['colorscale'] = colors
+    return info
+
+
+def _fill_html_template(info, embed_js=True):
+    as_json = json.dumps(info)
+    as_html = get_html_template('surface_plot_template.html').replace(
+        'INSERT_STAT_MAP_JSON_HERE', as_json)
+    as_html = add_js_lib(as_html, embed_js=embed_js)
+    return SurfaceView(as_html)
+
+
+def view_img_on_surf(stat_map_img, mesh='fsaverage5',
+                     threshold=None, cmap=cm.cold_hot,
+                     black_bg=False):
+    """
+    Insert a surface plot of a statistical map into an HTML page.
+
+    Parameters
+    ----------
+    stat_map_img : Niimg-like object, 3d
+        See http://nilearn.github.io/manipulating_images/input_output.html
+
+    mesh : str or dict, optional.
```

Comment:
> I am in favour of removing dictionary as input argument and use only strings. Because anyway we can choose this options once an image is opened in a browser. 

passing the mesh dictionary allows the user to choose which mesh corresponds to each of the keys ('pial_left', .. etc). For example, they might want to use a subject-specific cortical mesh rather than the default fsaverage ones. The menu in the browser allows the user to change between which surface is displayed, but not the underlying meshes, i.e. they can coose to display left hemisphere or right hemisphere, but cannot change the mesh that is used to display the left hemisphere.

> The only options which are missing when an image in a browser is opened are sulc_left and sulc_right. Is there any reason for not including these options ?

these are only used as background images for the thresholded part of the image. they are not displayed by themselves.

Reply:
> I think the doc string should say by default on which hemisphere it will display.

both will be available when the plot is opened

Reply:
> Side note: Code breaks when I gave a surface mesh in dictionary ? Could you please add tests if you consider keeping dictionary as input argument.

could you provide a snippet to reproduce? I added this test:

```
fsaverage = fetch_surf_fsaverage()
html = html_surface.view_img_on_surf(img, threshold=0, surf_mesh=fsaverage)
```

and it seems to pass. here fsaverage is a dictionary